src/screens/ManageFeedScreen.js

```diff
@@ -64,85 +63,101 @@ const styles = StyleSheet.create({
 
-    this.props = props
-    this.state = {
-      currData: GET_HOME_SECTIONS(props.publication),
-    }
     this.newOrder = null
     this.instructions =
       'Press down and drag the sections to the order you would like to see them appear on the home page'
   }
 
   componentDidMount() {
-    this.orderItems()
-    this.props.navigation.setParams({ handleSave: this._handleSave.bind(this) })
+    this.props.navigation.setParams({ handleSave: this._handleSave })
   }
 
   _handleSave = async () => {
     if (this.newOrder == null) return
     var sorted = []
     this.newOrder.forEach(i => {
-      this.state.currData.forEach((d, j) => {
+      this.currData.forEach((d, j) => {
         if (i == j) {
           sorted.push(d)
         }
       })
     })
     this.newData = sorted
-    if (this.newData == this.state.currData) return
+    if (this.newData == this.currData) return
     // console.log('saving-', this.newData)
-    await Storage.setItem(
+    let savedSuccessfully = await Storage.setItem(
       GET_HOME_FEED_ORDER_KEY(this.props.publication),
       this.newData
     )
 
-    this.props.dispatch({
-      type: REORDER_HOME_SECTIONS,
-      publication: this.props.publication,
-    })
-  }
-
-  orderItems = async () => {
-    let storedOrder = await Storage.getItem(
-      GET_HOME_FEED_ORDER_KEY(this.props.publication)
-    )
-    if (storedOrder != null) {
-      this.setState({ currData: storedOrder })
-    }
-  }
-
-  onReleaseRow = (key, currentOrder) => {
-    this.newOrder = currentOrder
+    if (savedSuccessfully)
+      this.props.dispatch(
+        updateHomeSections(this.props.publication, this.newData)
+      )
   }
 
   render() {
     return (
-      <View style={styles.container}>
+      <View style={styles.container} key={this.publication}>
         <View style={styles.sectionContainer}>
           <SortableList
             style={styles.list}
-            data={this.state.currData}
-            renderRow={this._renderRow}
-            onReleaseRow={this.onReleaseRow}
+            data={this.currData}
+            renderRow={({ data, active }) => {
+              return <Row data={data} active={active} />
+            }}
+            onReleaseRow={(key, currentOrder) => {
+              this.newOrder = currentOrder
+            }}
           />
         </View>
         <Text style={styles.description}>{this.instructions}</Text>
       </View>
     )
   }
+}
 
-  _renderRow = ({ data, active }) => {
-    return <Row data={data} active={active} />
-  }
+const ManageFeedScreenComp = ({
+  navigation,
+  publication,
+  settings,
+  dispatch,
+}) => {
+  // console.log('MANAGE FEED SCREEN COMP', publication)
+  return (
+    <ManageFeedForPublication
+      key={publication}
+      navigation={navigation}
+      publication={publication}
+      homeSectionPreferences={settings.homeSectionPreferences}
+      dispatch={dispatch}
+    />
+  )
 }
 
 class Row extends Component {
```

Comment:
convert to functional component

Reply:
going to leave this - will replace entire component in different PR